addons/project/models/project_project.py

```diff
@@ -434,7 +433,7 @@ def write(self, vals):
 
         res = super(Project, self).write(vals) if vals else True
 
-        if 'allow_recurring_tasks' in vals and not vals.get('allow_recurring_tasks'):
+        if not self.env.user.has_group('project.group_project_recurring_tasks'):
             self.env['project.task'].search([('project_id', 'in', self.ids), ('recurring_task', '=', True)]).write({'recurring_task': False})
```

Comment:
This code should be moved in `res.config.settings` instead of doing that each time we do a change in a project.